pkg/datapath/linux/node_ids.go

```diff
@@ -149,3 +216,51 @@ func (n *linuxNodeHandler) DumpNodeIDs() []*models.NodeID {
 	}
 	return dump
 }
+
+// RestoreNodeIDs restores node IDs and their associated IP addresses from the
+// BPF map and into the node handler in-memory copy.
+func (n *linuxNodeHandler) RestoreNodeIDs() {
+	// Retrieve node IDs from the BPF map to be able to restore them.
+	nodeIDs := make(map[string]uint16)
+	parse := func(key *nodemap.NodeKey, val *nodemap.NodeValue) {
+		address := key.IP.String()
+		if key.Family == bpf.EndpointKeyIPv4 {
+			address = net.IP(key.IP[:net.IPv4len]).String()
+		}
+		nodeIDs[address] = val.NodeID
+	}
+	if err := nodemap.NodeMap.IterateWithCallback(parse); err != nil {
+		log.WithError(err).Error("Failed to dump content of node map")
+		return
+	}
+
+	n.registerNodeIDAllocations(nodeIDs)
+	log.Infof("Restored %d node IDs from the BPF map", len(nodeIDs))
+}
+
+func (n *linuxNodeHandler) registerNodeIDAllocations(allocatedNodeIDs map[string]uint16) {
+	n.mutex.Lock()
+	defer n.mutex.Unlock()
+
+	if len(n.nodeIDsByIPs) > 0 {
+		// If this happens, we likely have a bug in the startup logic and
+		// restored node IDs too late (after new node IDs were allocated).
+		log.Error("The node manager already contains node IDs")
+	}
+
+	n.nodeIDsByIPs = allocatedNodeIDs
```

Comment:
Perhaps add a comment here on why we need to both set `n.nodeIDsByIPs` and remove the id from `n.nodeIDs`